gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/model/ImportedCell.java

```diff
@@ -33,11 +33,17 @@
 	@Setter
 	private String score;
 
+	@Getter
+	@Setter
+	private String rawScore;
```

Comment:
Can you explain this rawScore variable briefly?

Reply:
Yeah, so regardless of locale grades are saved in the DB in period decimal format. This value is what's stored in the `score' variable here (if the user is in a comma decimal locale, the value gets converted so that the saved value is in period decimal format).

When we introduced front-end grade validation, the GradeValidator originally got passed this `score` value for validation. This presented the problem that if you've imported a file in comma locale, and you have a grade that is technically invalid (ie., 5,5555 -> too many decimal places), the GradeValidation routine would not flag this as invalid because it was being passed the processed value of 5.55 rather than 5,5555.

So we needed another variable here to carry the original (unparsed, unformatted) data from the spreadsheet, so that the front-end grade validation would be performed on the input data, rather than the processed data.

Reply:
MAkes sense to me. thanks.